test/Microsoft.AspNetCore.Razor.Tools.Test/ServerLifecycleTest.cs

```diff
@@ -215,16 +223,18 @@ public void ServerStartup_SuccessfullyAcquiredMutex()
 
             using (var serverData = ServerUtilities.CreateServer(compilerHost: host))
             {
-                var tasks = new List<Task<ServerResponse>>();
+                // Send all the requests.
+                var clients = new List<Client>();
                 for (var i = 0; i < requestCount; i++)
                 {
-                    var task = ServerUtilities.Send(serverData.PipeName, EmptyServerRequest);
-                    tasks.Add(task);
+                    var client = await Client.ConnectAsync(serverData.PipeName, timeout: null, cancellationToken: default);
```

Comment:
This is kinda similar to the previous two tests. We want to make sure we send all the requests before we send the cancellation and read the response. Before this, we might try to read the response before the cancellation goes through.